### PR TITLE
[DROOLS-6646] Read/Write accessors cleanup

### PR DESCRIPTION
**Pull request by mariofusco**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6646

### CODE REVIEW

File: drools-core/src/main/java/org/drools/core/base/evaluators/Operator.java [hunk 1]
```diff
@@ -25,54 +25,70 @@
...
-    implements
-    Externalizable, org.kie.api.runtime.rule.Operator {
-
-    private static final long                  serialVersionUID = 510l;
+public class Operator implements Externalizable, org.kie.api.runtime.rule.Operator {
 
     // a static private cache so that pluggable operator can register their implementations
     // it is automatically initialized with common operator implementations
-    private static final Map<String, Operator> CACHE            = Collections.synchronizedMap( new HashMap<String, Operator>() );
+    private static final Map<String, Operator> CACHE = Collections.synchronizedMap( new HashMap<String, Operator>() );
```
Comment (danielezonca) [inline drools-core/src/main/java/org/drools/core/base/evaluators/Operator.java]: What it is different to use a `ConcurrentHashMap`?

File: kie-internal/src/main/java/org/kie/internal/utils/ChainedProperties.java [hunk 2]
```diff
@@ -83,6 +83,7 @@ private ChainedProperties(String confFileName, ClassLoader classLoader) {
 
         loadProperties( "META-INF/kie." + confFileName, classLoader, this.props );
         loadProperties( "META-INF/kie.default." + confFileName, classLoader, this.defaultProps);
+        loadProperties( "META-INF/kie.default.mvel." + confFileName, classLoader, this.defaultProps);
```
Comment (danielezonca) [inline kie-internal/src/main/java/org/kie/internal/utils/ChainedProperties.java:86]: How does this work?

File: drools-test-coverage/test-compiler-integration/src/test/java/org/drools/mvel/integrationtests/StrEvaluatorTest.java [hunk 3]
```diff
@@ -44,8 +44,7 @@ public StrEvaluatorTest(final KieBaseTestConfiguration kieBaseTestConfiguration)
 
     @Parameterized.Parameters(name = "KieBase type={0}")
     public static Collection<Object[]> getParameters() {
-     // TODO: EM failed with some tests. File JIRAs
-        return TestParametersUtil.getKieBaseCloudConfigurations(false);
+        return TestParametersUtil.getKieBaseCloudConfigurations(true);
```
Comment (danielezonca) [inline drools-test-coverage/test-compiler-integration/src/test/java/org/drools/mvel/integrationtests/StrEvaluatorTest.java:47]: Why this change? Is this PR changing some behaviour?

File: drools-core/src/main/java/org/drools/core/base/evaluators/Operator.java [hunk 4]
```diff
@@ -25,54 +25,70 @@
...
-    public static final Operator  SOUNDSLIKE       = Operator.addOperatorToRegistry( "soundslike",
-                                                                                     false );
-    public static final Operator  NOT_SOUNDSLIKE   = Operator.addOperatorToRegistry( "soundslike",
-                                                                                     true );
+    public static final Operator               EQUAL            = addOperatorToRegistry( "==", false );
+    public static final Operator               NOT_EQUAL        = addOperatorToRegistry( "!=", false );
+    public static final Operator               LESS             = addOperatorToRegistry( "<", false );
+    public static final Operator               LESS_OR_EQUAL    = addOperatorToRegistry( "<=", false );
+    public static final Operator               GREATER          = addOperatorToRegistry( ">", false );
+    public static final Operator               GREATER_OR_EQUAL = addOperatorToRegistry( ">=", false );
```
Comment (danielezonca) [inline drools-core/src/main/java/org/drools/core/base/evaluators/Operator.java]: What is the difference between these operators and the `Op` enum?

File: drools-core/src/main/java/org/drools/core/base/evaluators/Operator.java (hunk 1 above)
Comment (mariofusco) [reply drools-core/src/main/java/org/drools/core/base/evaluators/Operator.java]: Good suggestion. I just changed the formatting without checking it. I'll replace it with a `ConcurrentHashMap`.

File: kie-internal/src/main/java/org/kie/internal/utils/ChainedProperties.java (hunk 2 above)
Comment (mariofusco) [reply kie-internal/src/main/java/org/kie/internal/utils/ChainedProperties.java:86]: I needed another conf file in the `drools-mvel` module since the mvel evaluator have been moved there. I could have used the prefix `kie.default.` also for that one, but I wanted to avoid having the same file with the same name and position in 2 different modules to prevent the problems with uber-jar creation that we had in past.

File: drools-test-coverage/test-compiler-integration/src/test/java/org/drools/mvel/integrationtests/StrEvaluatorTest.java (hunk 3 above)
Comment (mariofusco) [reply drools-test-coverage/test-compiler-integration/src/test/java/org/drools/mvel/integrationtests/StrEvaluatorTest.java:47]: My refactor caused a problem with the `StrEvaluator` and I used this test to debug it. Then I realized that it was working also with the execmodel (nothing related with my refactor I think, I believe it worked even before but we didn't realize it) and then I enabled it.

File: drools-core/src/main/java/org/drools/core/base/evaluators/Operator.java (hunk 4 above)
Comment (mariofusco) [reply drools-core/src/main/java/org/drools/core/base/evaluators/Operator.java]: To be honest I don't know, I left them there as they were before my refactor, but indeed it would make sense to unify all the operators including them. I will try to also move them into the enum.

**Review by danielezonca — APPROVED** [review]
LGTM 👍
